docs/features/incremental-generators.cookbook.md

```diff
@@ -636,7 +636,248 @@ TODO: https://github.com/dotnet/roslyn/issues/72149
+                .GetMembers()
+                .OfType<IPropertySymbol>()
+                .ToArray();
+
+            InterfacePropertyModel[] propertyModels = interfaceProperties
+                .Select(interfaceProperty =>
+                {
+                    /*Using "interfaceProperty.Type" instead of "interfaceProperty.Type.Name" in order to not have error in specific cases.
+                    Es. the type "int" has Name "Int32", but writing "Int32" casue compilation error if we are not using "System" namespace*/
+                    string type = interfaceProperty.Type.ToString();
+                    return new InterfacePropertyModel(type, interfaceProperty.Name, interfaceProperty.SetMethod is not null);
+                })
+                .ToArray();
+
+            string containingNamespace = interfaceSymbol.ContainingNamespace.ToString();
+
+            ret.Add(new InterfaceModel(interfaceName, containingNamespace, interfaceUsings, propertyModels));
+        }
+
+        return [.. ret];
+    }
+    private static string[] GetUsings(SyntaxNode root)
+    {
+        return root.DescendantNodes()
+                .OfType<UsingDirectiveSyntax>()
+                .Select(usingDirective =>
+                {
+                    //get string "System" from string "using System;"
+                    string usingName = usingDirective.ToString().Split(' ').Last().TrimEnd(';');
+                    return usingName;
+                })
+                .ToArray();
+    }
+    private static string[] GetInterfacesNames(SyntaxList<AttributeListSyntax> attributeLists)
+    {
+        foreach (AttributeListSyntax attributeList in attributeLists)
+        {
+            foreach (AttributeSyntax attribute in attributeList.Attributes)
+            {
+                if (!attribute.Name.ToString().Equals(AutoImplementAttributeName) || attribute.ArgumentList is null)
+                    continue;
+
+                return attribute.ArgumentList.Arguments
+                    .Select(x => GetInterfaceName(x.Expression.ToString()))
+                    .ToArray();
+            }
+        }
+
+        return [];
+    }
+    private static string GetInterfaceName(string expression)
+    {
+        // Empty String
+        if (string.IsNullOrWhiteSpace(expression))
+            return expression;
+
+        // Interface Name got by nameof()
+        if (expression.StartsWith(NameOfString))
+            return expression.Substring(NameOfString.Length, expression.Length - NameOfString.Length - 1);
+
+        //Trimming ""
+        string ret = expression.Trim('"');
+        return ret;
+    }
+    private static INamedTypeSymbol? GetInterfaceSymbol(Compilation compilation, string[] nameSpaces, string implementedInterfaceName)
+    {
+        // Try Get without NameSpace if interface was already Fully Qualified
+        INamedTypeSymbol? interfaceSymbol = compilation.GetTypeByMetadataName(implementedInterfaceName);
```

Comment:
You don't need to do this. Change the `AutoImplement` attribute ctor to be `Type[]` instead of `string[]`, and then just leverage the attribute arguments which will already be types. It's both more robust doesn't need the lookup mechanism you're using here, which will likely fail.